spring-batch-infrastructure/src/test/java/org/springframework/batch/support/transaction/ResourcelessTransactionManagerTests.java

```diff
@@ -161,71 +132,55 @@ public void afterCompletion(int status) {
-					TransactionSynchronizationManager.registerSynchronization(new TransactionSynchronizationAdapter() {
-                        @Override
-						public void afterCompletion(int status) {
-							super.afterCompletion(status);
-							txStatus = status;
-							count++;
-						}
-					});
-					transactionTemplate.execute(new TransactionCallback<Void>() {
-                        @Override
-						public Void doInTransaction(TransactionStatus status) {
-							assertEquals(0, count);
-							count++;
-							throw new RuntimeException("Rollback!");
-						}
-					});
-					assertEquals(1, count);
-					return null;
-				}
+			transactionTemplate.execute(outerStatus -> {
+				TransactionSynchronizationManager.registerSynchronization(new TransactionSynchronization() {
+					@Override
+					public void afterCompletion(int status) {
+						txStatus = status;
+						count++;
+					}
+				});
+				transactionTemplate.execute(innerStatus -> {
+					assertEquals(0, count);
+					count++;
+					throw new RuntimeException("Rollback!");
+				});
+				assertEquals(1, count);
+				return null;
 			});
 			fail("Expected RuntimeException");
-		}
-		catch (RuntimeException e) {
+		} catch (RuntimeException e) {
 			assertEquals("Rollback!", e.getMessage());
 		}
 		assertEquals(TransactionSynchronization.STATUS_ROLLED_BACK, txStatus);
 		assertEquals(2, count);
 	}
 
 	@Test
-	public void testRollbackNestedOuter() throws Exception {
+	public void testRollbackNestedOuter() {
 		final TransactionTemplate transactionTemplate = new TransactionTemplate(transactionManager);
 		try {
-			transactionTemplate.execute(new TransactionCallback<Void>() {
-                @Override
-				public Void doInTransaction(TransactionStatus status) {
-					TransactionSynchronizationManager.registerSynchronization(new TransactionSynchronizationAdapter() {
-                        @Override
-						public void afterCompletion(int status) {
-							super.afterCompletion(status);
-							txStatus = status;
-							count++;
-						}
-					});
-					transactionTemplate.execute(new TransactionCallback<Void>() {
-                        @Override
-						public Void doInTransaction(TransactionStatus status) {
-							assertEquals(0, count);
-							count++;
-							return null;
-						}
-					});
-					assertEquals(1, count);
-					throw new RuntimeException("Rollback!");
-				}
+			transactionTemplate.execute(outerStatus -> {
+				TransactionSynchronizationManager.registerSynchronization(new TransactionSynchronization() {
+					@Override
+					public void afterCompletion(int status) {
+						txStatus = status;
+						count++;
+					}
+				});
+				transactionTemplate.execute(innerStatus -> {
+					assertEquals(0, count);
+					count++;
+					return null;
+				});
+				assertEquals(1, count);
+				throw new RuntimeException("Rollback!");
 			});
 			fail("Expected RuntimeException");
-		}
-		catch (RuntimeException e) {
+		} catch (RuntimeException e) {
```

Comment:
Please note that we need to follow the [same code formatting as Spring Framework](https://github.com/spring-projects/spring-batch/blob/master/CONTRIBUTING.md#code-style), which uses the [K&R style](https://github.com/spring-projects/spring-framework/wiki/Code-Style#block-like-constructs-kr-style). This is not an issue here, I will revert it on merge. I'm just mentioning it so that you are aware of that.

Reply:
Reverted the changed style for the catch blocks.